packages/meditrak-app-server/src/routes/sync/PullChangesRoute.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2022 Beyond Essential Systems Pty Ltd
+ */
+
+import { Request } from 'express';
+
+import { TYPES } from '@tupaia/database';
+import { Route } from '@tupaia/server-boilerplate';
+import { DatabaseError } from '@tupaia/utils';
+import { getChangesFilter } from './getChangesFilter';
+import { getUnsupportedModelFields } from '../../sync';
+
+type ChangeRecord = {
+  action: 'update' | 'delete';
+  recordType: string;
+  timestamp: number;
+  record?: Record<string, unknown>;
+  error?: { error: string };
+};
+
+export type PullChangesRequest = Request<
+  Record<string, never>,
+  ChangeRecord[],
+  Record<string, unknown>,
+  { appVersion: string; since?: string; recordTypes?: string; limit?: string; offset?: string }
+>;
+
+const MAX_CHANGES_RETURNED = 100;
+
+async function filterNullProperties(record: Record<string, unknown>) {
+  const recordWithoutNulls: Record<string, unknown> = {};
+  // Remove null entries to a) save bandwidth and b) remain consistent with previous mongo based db
+  // which simply had no key for undefined properties, whereas postgres uses null
```

Comment:
Ahaha so funny how old some of these comments are... "previous mongo based db" 😂

Reply:
`Realm` is based on mongo db yeah? Hehehe so are we still based on it? 😉

Reply:
Ha! Well kinda...this is the timeline of births:
- Mongo db is born
- In an unrelated company, Realm is born, a cool new database built and optimised for mobile, completely from scratch, and providing a reactive programming model
- Tupaia is born, using realm (still not based on mongo db) on the meditrak app, and mongo db on the server
- Tupaia flips to using postgres on the server, but has to support the old null behaviour
- Mongo buys realm and acts like it's based off the same tech 🙄